### PR TITLE
Fixing Javadoc build failures and checkstyle inconsistencies.

### PR DESCRIPTION
**Pull request by willsalz**
We're preparing to release v0.1.2, our first release to the central
repositories, which means we're going through a lot of new build phases
e.g. javadocs, gpg signing, maven releaseing, etc. I'm learning a lot
and hopefully we'll nail-down a quasi-sane workflow. But until then,
it's a lot of trial-and-error followed by frantic commits.

cc @brandoniles @truncs

### CODE REVIEW

File: tchannel-benchmark/src/main/java/com/uber/tchannel/benchmarks/PingPongServerBenchmark.java [hunk 1]
```diff
@@ -57,7 +57,11 @@
     TChannel client;
     int port;
 
-    @Param({ "0", "1", "10" })
+    @Param({
```
Comment (truncs) [inline tchannel-benchmark/src/main/java/com/uber/tchannel/benchmarks/PingPongServerBenchmark.java]: what are these changes for ?

File: tchannel-benchmark/src/main/java/com/uber/tchannel/benchmarks/PingPongServerBenchmark.java [hunk 2]
```diff
@@ -108,6 +112,21 @@ public void teardown() throws Exception {
         this.channel.shutdown();
     }
 
+    @AuxCounters
```
Comment (truncs) [inline tchannel-benchmark/src/main/java/com/uber/tchannel/benchmarks/PingPongServerBenchmark.java]: is this a style violation ?

File: tchannel-core/src/main/java/com/uber/tchannel/api/DefaultRequestHandler.java [hunk 3]
```diff
@@ -26,8 +26,10 @@
 
 public abstract class DefaultRequestHandler<T, U> implements RequestHandler<T, U> {
 
-    private TypeToken<T> requestType = new TypeToken<T>(getClass()) { };
-    private TypeToken<U> responseType = new TypeToken<U>(getClass()) { };
+    private TypeToken<T> requestType = new TypeToken<T>(getClass()) {
```
Comment (truncs) [inline tchannel-core/src/main/java/com/uber/tchannel/api/DefaultRequestHandler.java]: As long as we conform the automated style checks I don't think we should be making these changes. Its just not scalable, so either push these stylechecks to the automated ones or lets not fix them.

**Comment by truncs** [comment]
@willsalz why don't we just stay on 0.1.2-SNAPSHOT ? The project still looks like its in alpha phase.

**Comment by willsalz** [comment]
Yeah I can remove the auto-formatting from the diff.

And in terms of the version number: in order to publish to the central maven repository we have to push non-SNAPSHOT builds **unless** we have our clients put special declarations in their pom.xml stating specifically *which* SNAPSHOT repository to allow ( in this case, Sonatype's ). So we can keep a 'zero-major' SemVer version e.g. pre-1.0 and deploy those with standard machinery common to uploading projects to the repo.

**Comment by truncs** [comment]
that is why I like the netty style versioning better where each release is tagged with either Alpha, Beta or Final. Or we can just push the candidate to internal artifactory but not upload it to maven until its stable enough for general use.

**Comment by willsalz** [comment]
Merging to unblock.

**Pull request merged**